crypto/ec/ec_mult.c

```diff
@@ -282,14 +302,15 @@ int ossl_ec_scalar_mul_ladder(const EC_GROUP *group, EC_POINT *r,
     /* top bit is a 1, in a fixed pos */
     pbit = 1;
 
-#define EC_POINT_CSWAP(c, a, b, w, t) do {         \
-        BN_consttime_swap(c, (a)->X, (b)->X, w);   \
-        BN_consttime_swap(c, (a)->Y, (b)->Y, w);   \
-        BN_consttime_swap(c, (a)->Z, (b)->Z, w);   \
-        t = ((a)->Z_is_one ^ (b)->Z_is_one) & (c); \
-        (a)->Z_is_one ^= (t);                      \
-        (b)->Z_is_one ^= (t);                      \
-} while(0)
+#define EC_POINT_CSWAP(c, a, b, w, t, r) \
+    do {                                                          \
+        BN_consttime_randomized_swap(c, (a)->X, (b)->X, w, r, r); \
+        BN_consttime_randomized_swap(c, (a)->Y, (b)->Y, w, r, r); \
+        BN_consttime_randomized_swap(c, (a)->Z, (b)->Z, w, r, r); \
+        t = (((a)->Z_is_one ^ (b)->Z_is_one) & (c)) ^ (r);        \
+        (a)->Z_is_one = ((a)->Z_is_one ^ (t)) ^ (r);              \
+        (b)->Z_is_one = ((b)->Z_is_one ^ (t)) ^ (r);              \
```

Comment:
these extra terms `^ (r)` will most likely optimized away.
you should use a value_barrier here see 
https://github.com/openssl/openssl/blob/e59bfbaa2dbd680f77e1121e382502bd522a466c/include/internal/constant_time.h#L229-L238

Reply:
e.g.
```C
  t = (...) ^ (r);
  t = value_barrier(t);
  (a)->Z_is_one = ((a)->Z_is_one ^ (t)) ^ (r);
```